common/darwin/Classes/FlutterRTCMediaStream.m

```diff
@@ -494,28 +494,84 @@ -(void)createLocalMediaStream:(FlutterResult)result{
 }
 
 -(void)getSources:(FlutterResult)result{
-  NSMutableArray *sources = [NSMutableArray array];
-  NSArray *videoDevices = [AVCaptureDevice devicesWithMediaType:AVMediaTypeVideo];
-  for (AVCaptureDevice *device in videoDevices) {
-    [sources addObject:@{
-                         @"facing": device.positionString,
-                         @"deviceId": device.uniqueID,
-                         @"label": device.localizedName,
-                         @"kind": @"videoinput",
-                         }];
-  }
-  NSArray *audioDevices = [AVCaptureDevice devicesWithMediaType:AVMediaTypeAudio];
-  for (AVCaptureDevice *device in audioDevices) {
-    [sources addObject:@{
-                         @"facing": @"",
-                         @"deviceId": device.uniqueID,
-                         @"label": device.localizedName,
-                         @"kind": @"audioinput",
-                         }];
-  }
+    NSMutableArray *sources = [NSMutableArray array];
+    NSArray *videoDevices = [AVCaptureDevice devicesWithMediaType:AVMediaTypeVideo];
+    for (AVCaptureDevice *device in videoDevices) {
+      [sources addObject:@{
+                           @"facing": device.positionString,
+                           @"deviceId": device.uniqueID,
+                           @"label": device.localizedName,
+                           @"kind": @"videoinput",
+                           }];
+    }
+#if TARGET_OS_IPHONE
+    AVAudioSession *session = [AVAudioSession sharedInstance];
```

Comment:
Why are you using direct `AVAudioSession` manipulation for iPhoneOS? WebRTC provided extended wrapper `RTCAudioSession` which allow to use [`useManualAudio`](https://github.com/webrtc-sdk/webrtc/blob/a828846eb98e404bf3b00c1abf3787f93171a39c/sdk/objc/components/audio/RTCAudioSession.h#L152) WebRTC functionality for correct integration with CallKit.

Reply:
yeah, You are right, we should change to RTCAudioSession